sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp

```diff
@@ -107,13 +110,14 @@ namespace Azure { namespace Core { namespace Http {
     bool NoSignal = false;
 
     /**
-     * @brief Contain the maximum time in seconds that you allow the connection phase to the server
-     * to take.
+     * @brief Contain the maximum time that you allow the connection phase to the server to take.
      *
      * @details This only limits the connection phase, it has no impact once it has connected.
      *
+     * @remarks The default timeout is 300 seconds and using `0` would set this default value.
+     *
      */
-    long ConnectionTimeout = DefaultConnectionTimeout;
+    std::chrono::milliseconds ConnectionTimeout = _detail::DefaultConnectionTimeout;
```

Comment:
@antkmsft you had recommended to use `std::chrono::milliseconds` instead of `long`. It generally makes sense to me, but just curious, what's your rationale for this?

Reply:
I am asking because the curl options use `long`. Should we deviate from that?

https://curl.se/libcurl/c/CURLOPT_CONNECTTIMEOUT_MS.html

@JeffreyRichter and @CaseyCarter any thoughts on the type here? Should we try to match the underlying curl library, or is using chrono milliseconds more idiomatic to C++?

Reply:
It's a win for customers.  They don't need to worry or be confused about what unit is the `long` representing.
Customers can also convert from minutes, seconds to milliseconds like
options.ConnectionTimeout = std::chrono::minutes(3);  ( this would set the option to 1000 * 60 * 3 )  // See the test I updated. Customer no longer need to convert to milliseconds exactly